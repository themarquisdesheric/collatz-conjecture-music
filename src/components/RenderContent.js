import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import CollatzGraph from './CollatzGraph';
import IntroBlurb from './IntroBlurb';

const RenderContent = ({ sequence, wave }) => (
  <Fragment>
    <main>
      {!!sequence.length
        ? <CollatzGraph sequence={sequence} wave={wave} />
        : <IntroBlurb />
      }
    </main>
    {!sequence.length && 
      <Fragment>
        <hr />
        <p className="pattern-sound">
          But how might the resulting pattern sound?<br />
          <span className="thinking-emoji" role="img" aria-label="thinking emoji">
            ðŸ¤”
          </span>
        </p>
      </Fragment>
    }
  </Fragment>
);

RenderContent.propTypes = {
  sequence: PropTypes.arrayOf(
    PropTypes.number
  ).isRequired,
  wave: PropTypes.string.isRequired
};

export default RenderContent;
