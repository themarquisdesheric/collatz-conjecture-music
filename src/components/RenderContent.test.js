import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import RenderContent from './RenderContent';

jest.mock('chart.js');

describe('renderContent', () => {
  const props = {
    sequence: [],
    wave: 'sine',
    isPlaying: false,
    handleCollatz: jest.fn(),
    handleWave: jest.fn(),
    handleRepeat: jest.fn()
  }
  
  let wrapper = mount(<RenderContent {...props} />);
  
  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  
  it('should render playing CollatzGraph if sequence.length is true', () => {
    props.sequence= [1];

    wrapper = mount(<RenderContent {...props} />);

    expect(wrapper.find('CollatzGraph').length).toBe(1);
    expect(wrapper.find('.thinking-emoji').length).toBe(0);
  });
});