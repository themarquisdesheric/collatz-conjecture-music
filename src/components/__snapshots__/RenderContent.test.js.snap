// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderContent should match snapshot 1`] = `
<RenderContent
  handleCollatz={[MockFunction]}
  handleRepeat={[MockFunction]}
  handleWave={[MockFunction]}
  isPlaying={false}
  sequence={Array []}
  wave="sine"
>
  <main>
    <IntroBlurb>
      <div
        className="intro-blurb"
      >
        <ul>
          <li>
            The
            <a
              href="https://en.wikipedia.org/wiki/Collatz_conjecture"
              rel="noopener noreferrer"
              target="_blank"
            >
              Collatz Conjecture
            </a>
            is an unsolved problem in mathematics.
          </li>
          <li>
            It states that given any positive integer, the sequence will always reach 1 by following two rules:
          </li>
          <li>
            If the integer is even,
            <span
              className="odd"
            >
              divide it by 2
            </span>
          </li>
          <li>
            Otherwise
            <span
              className="odd"
            >
              multiply it by 3
            </span>
            and
            <span
              className="odd"
            >
              add 1
            </span>
          </li>
          <li>
            Then repeat the process until you reach 1
          </li>
        </ul>
      </div>
    </IntroBlurb>
  </main>
  <hr />
  <p
    className="pattern-sound"
  >
    But how might the resulting pattern sound?
    <br />
    <span
      aria-label="thinking emoji"
      className="thinking-emoji"
      role="img"
    >
      ðŸ¤”
    </span>
  </p>
</RenderContent>
`;
